caterva2/tools/cat2_to_hdf5.py

```diff
@@ -0,0 +1,159 @@
+###############################################################################
+# Caterva2 - On demand access to remote Blosc2 data repositories
+#
+# Copyright (c) 2023 The Blosc Developers <blosc@blosc.org>
+# https://www.blosc.org
+# License: GNU Affero General Public License v3.0
+# See LICENSE.txt for details about copyright and rights to use.
+###############################################################################
+
+"""Export a Caterva2 root to datasets in an HDF5 file.
+
+The content of the existing Caterva2 root must be readable, while the HDF5
+file is created anew (it must not exist).  For each directory under the
+Caterva2 root, a group with the same path name is created in the HDF5 file.
+For each file or Blosc2 dataset, an HDF5 dataset with the same path name
+(minus any ``.b2*`` extension) is created in the HDF5 file.
+
+Blosc2 arrays (``*.b2nd``), frames (``*.b2frame``) and compressed files
+(``*.b2``) have their chunk sizes respected during conversion, while plain
+files use a default chunk size.  Arrays result in typed multidimensional
+datasets, while the rest result in flat datasets of bytes.  None of the
+exported datasets use compression yet.
+
+Warning: For the moment, the data in each file and dataset is read and
+decompressed into memory in its entirety.
+
+Datasets or attributes which are unsupported or fail to be converted are
+simply reported and skipped, and they do not cause the program to fail.
+"""
+
+import logging
+import os
+import pathlib
+import sys
+
+import blosc2
+import h5py
+import numpy
+
+from collections.abc import Callable, Iterator, Mapping
+
+
+def export_dataset(c2_leaf: os.DirEntry, h5_group: h5py.Group,
+                   read: Callable[[str], tuple[Mapping, Mapping]]) -> None:
+    try:
+        (kwds, attrs) = read(c2_leaf.path)
+    except Exception as e:
+        logging.error(f"Failed to read Blosc2 dataset: "
+                      f"{c2_leaf.path!r} -> {e!r}")
+        return
+
+    try:
+        h5_dataset = h5_group.create_dataset(**kwds)
```

Comment:
It would be nice to use a compressor by default when exporting.  As we are using Blosc2, my suggestion would be to import hdf5plugin and use Blosc2 with values like `cname='zstd', clevel=5, filters=1` (see http://www.silx.org/doc/hdf5plugin/latest/usage.html#hdf5plugin.Blosc2).  BTW, this naming schema for compression params is inherited from Blosc1, and it could change in the future (but we can adapt to this later on).

Reply:
Thanks! Commit 007ee1de sets the default compression parameters that you suggested.